pkg/innerring/innerring.go

```diff
@@ -388,6 +394,48 @@ func New(ctx context.Context, log *zap.Logger, cfg *viper.Viper) (*Server, error
 		return nil, err
 	}
 
+	cnrClient, err := invoke.NewNoFeeContainerClient(server.morphClient, server.contracts.container)
+	if err != nil {
+		return nil, err
+	}
+
+	nmClient, err := invoke.NewNoFeeNetmapClient(server.morphClient, server.contracts.netmap)
+	if err != nil {
+		return nil, err
+	}
+
+	balClient, err := invoke.NewNoFeeBalanceClient(server.morphClient, server.contracts.balance)
+	if err != nil {
+		return nil, err
+	}
+
+	auditCalcDeps := &auditSettlementDeps{
+		log:           server.log,
+		cnrSrc:        cnrClient,
+		auditClient:   server.auditClient,
+		nmSrc:         nmClient,
+		clientCache:   clientCache,
+		balanceClient: balClient,
+	}
+
+	server.auditSettlement = auditSettlement.NewCalculator(
+		&auditSettlement.CalculatorPrm{
+			ResultStorage:       auditCalcDeps,
+			ContainerStorage:    auditCalcDeps,
+			PlacementCalculator: auditCalcDeps,
+			SGStorage:           auditCalcDeps,
+			AccountStorage:      auditCalcDeps,
+			Exchanger:           auditCalcDeps,
+		},
+		auditSettlement.WithLogger(server.log),
+	)
+
+	server.subscribeNewEpoch(func(e netmapEvent.NewEpoch) {
```

Comment:
I would like to start all actions related to new epoch in one place -- netmap processor. One notification event has exactly one handler, keep it simple as that. 

So I suggest to expand netmap processor and start audit settlement in `processNewEpoch` function.
```go
func (np *Processor) processNewEpoch(epoch uint64) {
	...
	np.netmapSnapshot.update(snapshot, epoch)
	np.handleCleanupTick(netmapCleanupTick{epoch: epoch})
	np.handleNewAudit(audit.NewAuditStartEvent(epoch))
        
        np.handleAuditSettlement(&auditSettlement.CalculatePrm(epoch)) // something like that?
}

Related to this: settlement package does not contain processor right now. Maybe it will be nice to have one and implement start of audit settlement as an event handling (the same way new audit starts). I can implement this in separate PR after this one will be merged.

Reply:
I will try to implement it as a finishing touch.